nano/node/node.cpp

```diff
@@ -1182,12 +1189,12 @@ void nano::node::start_election (std::shared_ptr<nano::block> const & block)
 bool nano::node::block_confirmed (nano::block_hash const & hash_a)
 {
 	auto transaction (store.tx_begin_read ());
-	return ledger.block_confirmed (transaction, hash_a);
+	return ledger.confirmed.exists_or_pruned (transaction, hash_a);
 }
 
 bool nano::node::block_confirmed_or_being_confirmed (nano::store::transaction const & transaction, nano::block_hash const & hash_a)
 {
-	return confirming_set.exists (hash_a) || ledger.block_confirmed (transaction, hash_a);
+	return confirming_set.exists (hash_a) || ledger.confirmed.exists_or_pruned (transaction, hash_a);
```

Comment:
With the way it's implemented now, a big warning should be put there. In LMDB write transactions do not block readers, nor do readers block writers. So it's possible that a read transaction is started with block being unconfirmed, then confirming set starts confirming and successfully processes and flushes it (`confirming_set.exists (...)` returns false) and a subsequent check `ledger.confirmed.exists_or_pruned ()` still returns false because read transaction was started before the block was confirmed.

Reply:
Perhaps this could accept a write transaction, however this could introduce a significant performance degradation, depending on how it's used.

Reply:
This is a good point, I didn't see that race condition before.  Since this isn't new behavior I opened a separate PR for adding the comment. https://github.com/nanocurrency/nano-node/pull/4540